fastchat/train/llama_flash_attn_monkey_patch.py

```diff
@@ -61,54 +50,39 @@ def forward(
         query_states, key_states, cos, sin, position_ids
     )
 
-    # Past Key value support
     if past_key_value is not None:
-        # reuse k, v, self_attention
+        # reuse k, v
         key_states = torch.cat([past_key_value[0], key_states], dim=2)
         value_states = torch.cat([past_key_value[1], value_states], dim=2)
 
     past_key_value = (key_states, value_states) if use_cache else None
 
-    # Flash attention codes from
-    # https://github.com/HazyResearch/flash-attention/blob/main/flash_attn/flash_attention.py
-
-    # transform the data into the format required by flash attention
-    qkv = torch.stack(
-        [query_states, key_states, value_states], dim=2
-    )  # [bsz, nh, 3, q_len, hd]
-    qkv = qkv.transpose(1, 3)  # [bsz, q_len, 3, nh, hd]
-    # We have disabled _prepare_decoder_attention_mask in LlamaModel
-    # the attention_mask should be the same as the key_padding_mask
```

Comment:
this note about how `_prepare_decoder_attention_mask` has been neutered seems relevant? i think that's important to keep in place

Reply:
Okay, we can add it back.